### DISCUSSION TITLE
Unable to resolve link; aborting

### DISCUSSION THREAD

**Discussion by buhtz** [kaushalmodi/ox-hugo#586] [Q&A] [answered]
Hello I am getting the message `Unable to resolve link; aborting: "roam:git"` when using `org-hugo-export-wim-to-md`. See the full debug output at the end.

In the context of org-roam especially when used for Zettelkasten it is usual to have dead links. A dead link is not an error just an information because it will remind you that there is something to do ("thinking further").

From `ox-publish` I know the variable `(setq org-export-with-broken-links t)`. It seems to me that this is not relevant for `ox-hugo`, correct?
I did `M-x customize-group RET ox-hugo` but couldn't see an option like that there.

Can ox-hugo export such links? It could just replace them with the string `ERROR`or whatever it wants. But it shouldn't stop the export process.

```emacs-lisp
  1 Debugger entered--Lisp error: (org-link-broken "roam:git")
  2   signal(org-link-broken ("roam:git"))
  3   org-export-resolve-fuzzy-link((link (:type "fuzzy" :path "roam:git" :format bracket :raw-link "roam:git" :appli$
  4   #f(compiled-function (link) #<bytecode 0x158ff9168c05>)((link (:type "fuzzy" :path "roam:git" :format bracket :$
  5   #f(compiled-function (--data) #<bytecode 0x158ff9168c65>)((link (:type "fuzzy" :path "roam:git" :format bracket$
  6   mapc(#f(compiled-function (--data) #<bytecode 0x158ff9168c65>) (#("SSH-Key for Codeberg.org erstellen, um ohne $
  7   #f(compiled-function (--data) #<bytecode 0x158ff9168c65>)((paragraph (:begin 155 :end 294 :contents-begin 155 :$
  8   mapc(#f(compiled-function (--data) #<bytecode 0x158ff9168c65>) ((paragraph (:begin 155 :end 294 :contents-begin$
  9   #f(compiled-function (--data) #<bytecode 0x158ff9168c65>)((section (:begin 155 :end 295 :contents-begin 155 :co$
 10   mapc(#f(compiled-function (--data) #<bytecode 0x158ff9168c65>) ((section (:begin 155 :end 295 :contents-begin 1$
 11   #f(compiled-function (--data) #<bytecode 0x158ff9168c65>)((headline (:raw-value "Ziel" :begin 148 :end 295 :pre$
 12   mapc(#f(compiled-function (--data) #<bytecode 0x158ff9168c65>) ((section (:begin 1 :end 148 :contents-begin 1 :$
 13   #f(compiled-function (--data) #<bytecode 0x158ff9168c65>)((org-data nil (section (:begin 1 :end 148 :contents-b$
 14   org-element-map((org-data nil (section (:begin 1 :end 148 :contents-begin 1 :contents-end 148 :post-blank 0 :po$
 15   org-hugo--get-pre-processed-buffer()
 16   org-hugo-export-wim-to-md(:all-subtrees)
 17   (save-current-buffer (set-buffer (find-file-noselect org-file)) (message (format "[ox-hugo/export-all file %d/%$
 18   (let ((org-file (car --dolist-tail--))) (save-current-buffer (set-buffer (find-file-noselect org-file)) (messag$
 19   (while --dolist-tail-- (let ((org-file (car --dolist-tail--))) (save-current-buffer (set-buffer (find-file-nose$
 20   (let ((--dolist-tail-- org-files)) (while --dolist-tail-- (let ((org-file (car --dolist-tail--))) (save-current$
 21   (progn (message (format (if dont-recurse "[ox-hugo/export-all] Exporting %d files from %S .." "[ox-hugo/export-$
 22   (if (= 0 num-files) (message (format "No Org files found in %s" search-path)) (progn (message (format (if dont-$
 23   (let* ((org-files-root-dir (or org-files-root-dir default-directory)) (dont-recurse (or dont-recurse (and curre$
 24   ox-hugo/export-all("~/tab-cloud/my.org-roam")
 25   eval((ox-hugo/export-all "~/tab-cloud/my.org-roam") t)
 26   eval-expression((ox-hugo/export-all "~/tab-cloud/my.org-roam") nil nil 127)
 27   funcall-interactively(eval-expression (ox-hugo/export-all "~/tab-cloud/my.org-roam") nil nil 127)
 28   call-interactively(eval-expression nil nil)
 29   command-execute(eval-expression)
```

**Comment by kaushalmodi** [comment]
> I know the variable (setq org-export-with-broken-links t). It seems to me that this is not relevant for ox-hugo, correct?

Have you tried setting it to t?

From the error backtrace, that error is thrown by `org-export-resolve-fuzzy-link`, not `ox-hugo`. That function is part of the core ox.el. 

One way to check is to try to export the same to ox-html; I think you will see the same error there too. 

---

I believe there are quite a few org-roam users using ox-hugo. Try asking on org-roam forum (or emacs subreddit) on how they deal with the roam: links.

**Reply by buhtz** [reply]
> > I know the variable (setq org-export-with-broken-links t). It seems to me that this is not relevant for ox-hugo, correct?
> Have you tried setting it to t?

Yes I did and it is set (checked via `C-h v`).

> I believe there are quite a few org-roam users using ox-hugo. Try asking on org-roam forum (or emacs subreddit) on how they deal with the roam: links.

https://lists.gnu.org/archive/html/emacs-orgmode/2022-03/msg00133.html

**Reply by kaushalmodi** [reply]
Ok, do you see that error when you export using ox-md or ox-html?

Can you also share a minimal Org file here that can reproduce this error on an export?

**Reply by kaushalmodi** [reply]
@Codeberg-AsGithubAlternative-buhtz I have converted this to a new issue. You can track the issue's progress on https://github.com/kaushalmodi/ox-hugo/issues/587.

**Comment by kaushalmodi** [comment, marked as answer]
This is now fixed as long as `#+options: broken-links:t` keyword is used or `org-export-with-broken-links` is set to `t` . Ref https://github.com/kaushalmodi/ox-hugo/issues/587.
